### PR TITLE
Sort refactor and add test cases

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:

Refactored code based on Ivan comments.
Added test cases as suggested by Ivan and Val

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ChainedComparator.java [hunk 1]
```diff
@@ -22,6 +23,7 @@ public int compare(ReadDocument o1, ReadDocument o2) {
       if (compareValue != 0) return compareValue;
       i++;
     }
-    return 0;
+    return JsonNodeComparator.ascending()
```
[the remark this answers is not in the document]
Comment (tatu-at-datastax) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ChainedComparator.java:28]: Ok so ordering by primary key (_id), ascending?

[the remark this answers is not in the document]
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ChainedComparator.java:28]: If all the sort fields match, I added logic to sort ascending by document Id, just to make the result consistent.

**Review by tatu-at-datastax — APPROVED** [review]
Minor changes, but looks good otherwise.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java [hunk 2]
```diff
@@ -273,4 +267,17 @@ default Uni<CountResponse> countDocuments(
...
+
+  record DocJsonValue(ObjectMapper objectMapper, QueryOuterClass.Value docJsonValue)
+      implements Supplier<JsonNode> {
+    public JsonNode get() {
+      try {
+        return objectMapper.readTree(Values.string(docJsonValue));
+      } catch (JsonProcessingException e) {
+        // These are data stored in the DB so the error should never happen
+        // e.printStackTrace();
+        return null;
```
[the remark this answers is not in the document]
Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ReadOperation.java]: Will do

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ChainedComparator.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/ChainedComparator.java:28]: I am glad that at least one "bug" was uncovered by tests..

**Pull request merged**